lib/http.module.ts

```diff
@@ -1,114 +1,64 @@
-        },
-      ],
-    };
-  }
+      inject: [{ token: MODULE_OPTIONS_TOKEN, optional: true }],
+      useFactory: (options: HttpModuleOptions = {}) => {
+        const axiosInstance = Axios.create(options?.axiosConfig);
 
-  static registerAsync(options: HttpModuleAsyncOptions): DynamicModule {
-    return {
-      module: HttpModule,
-      imports: options.imports,
-      providers: [
-        ...this.createAsyncProviders(options),
-        {
-          provide: AXIOS_INSTANCE_TOKEN,
-          useFactory: (config: HttpModuleOptions) => createAxiosInstance(config),
-          inject: [HTTP_MODULE_OPTIONS],
-        },
-        {
-          provide: HTTP_MODULE_ID,
-          useValue: randomStringGenerator(),
-        },
-        ...(options.extraProviders || []),
-      ],
-    };
-  }
-
-  private static createAsyncProviders(
-      options: HttpModuleAsyncOptions,
-  ): Provider[] {
-    if (options.useExisting || options.useFactory) {
-      return [this.createAsyncOptionsProvider(options)];
-    }
+        axiosRetry(axiosInstance, options?.axiosRetryConfig);
 
-    const providers = [
-      this.createAsyncOptionsProvider(options)
-    ];
+        if (options.interceptors?.request) {
+          axiosInstance.interceptors.request.use(
+            options.interceptors.request.onFulfilled,
+            options.interceptors.request.onRejected,
+            options.interceptors.request.options,
+          );
+        }
 
-    if(options.useClass)
-      providers.push({
-        provide: options.useClass,
-        useClass: options.useClass,
-      })
-
-    return providers;
-  }
+        if (options.interceptors?.response) {
+          axiosInstance.interceptors.response.use(
+            options.interceptors.response.onFulfilled,
+            options.interceptors.response.onRejected,
+            options.interceptors.response.options,
+          );
+        }
 
-  private static createAsyncOptionsProvider(
-      options: HttpModuleAsyncOptions,
-  ): Provider {
-    if (options.useFactory) {
-      return {
-        provide: HTTP_MODULE_OPTIONS,
-        useFactory: options.useFactory,
-        inject: options.inject || [],
-      };
-    }
-
-    let inject;
-    if (options.useExisting)
-      inject = [options.useExisting];
-    else if (options.useClass)
-      inject = [options.useClass];
-
-    return {
-      provide: HTTP_MODULE_OPTIONS,
-      useFactory: async (optionsFactory: HttpModuleOptionsFactory) =>
-          optionsFactory.createHttpOptions(),
-      inject,
-    };
+        return axiosInstance;
+      },
+    },
+    {
+      provide: HTTP_MODULE_ID_TOKEN,
+      useValue: randomStringGenerator(),
+    },
+  ],
+  exports: [HttpService],
+})
+export class HttpModule extends ConfigurableModuleClass {
+  // Redefining the register static method to allow `undefined` options to be
+  // accepted by the type system
+  public static register(options?: HttpModuleOptions): DynamicModule {
```

Comment:
Why would you call register with undefined instead of just not calling register()?